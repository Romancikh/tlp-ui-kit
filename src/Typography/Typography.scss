.typography {
  margin: 0;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;

  &_variant {
    &_body2 {
      font-size: 0.875rem;
      line-height: 1.43;
      letter-spacing: 0.01071em;
    }

    &_button {
      font-weight: 500;
      font-size: 0.875rem;
      line-height: 1.75;
      letter-spacing: 0.02857em;
      text-transform: uppercase;
    }

    &_caption {
      font-size: 0.75rem;
      line-height: 1.66;
      letter-spacing: 0.03333em;
    }

    &_h1 {
      font-weight: 300;
      font-size: 6rem;
      line-height: 1.167;
      letter-spacing: -0.01562em;
    }

    &_h2 {
      font-weight: 300;
      font-size: 3.75rem;
      line-height: 1.2;
      letter-spacing: -0.00833em;
    }

    &_h3 {
      font-size: 3rem;
      line-height: 1.167;
      letter-spacing: 0em;
    }

    &_h4 {
      font-size: 2.125rem;
      line-height: 1.235;
      letter-spacing: 0.00735em;
    }

    &_h5 {
      font-size: 1.5rem;
      line-height: 1.334;
      letter-spacing: 0em;
    }

    &_h6 {
      font-weight: 500;
      font-size: 1.25rem;
      line-height: 1.6;
      letter-spacing: 0.0075em;
    }

    &_subtitle1 {
      line-height: 1.75;
    }

    &_subtitle2 {
      font-weight: 500;
      font-size: 0.875rem;
      line-height: 1.57;
      letter-spacing: 0.00714em;
    }
  }
}
