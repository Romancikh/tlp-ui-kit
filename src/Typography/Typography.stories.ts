import type { Meta, StoryObj } from "@storybook/react";

import Typography from "./Typography";

const meta = {
  title: "Typography",
  component: Typography,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Typography>;

export default meta;
type Story = StoryObj<typeof meta>;

export const h1: Story = {
  args: {
    variant: "h1",
    children: "h1. Heading",
  },
};

export const Colored: Story = {
  args: {
    color: "#1976d2",
    children:
      "body1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.",
  },
};
