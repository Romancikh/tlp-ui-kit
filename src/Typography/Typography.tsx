import { CSSProperties, PropsWithChildren, createElement } from "react";

import "./Typography.scss";

type TypographyProps = PropsWithChildren & {
  color?: string;
  fontSize?: number;
  fontWeight?: number;
  lineHeight?: number;
  marginLeft?: number;
  marginTop?: number;
  marginX?: number;
  marginY?: number;
  variant?:
    | "body1"
    | "body2"
    | "button"
    | "caption"
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "subtitle1"
    | "subtitle2";
  width?: number;
};

const component = {
  body1: "p",
  body2: "p",
  button: "span",
  caption: "span",
  h1: "h1",
  h2: "h2",
  h3: "h3",
  h4: "h4",
  h5: "h5",
  h6: "h6",
  subtitle1: "h6",
  subtitle2: "h6",
};

function Typography({
  children,
  color,
  fontSize,
  fontWeight,
  lineHeight,
  marginLeft,
  marginTop,
  marginX,
  marginY,
  variant = "body1",
  width,
}: TypographyProps) {
  const textStyle: CSSProperties = {
    color,
    fontSize,
    fontWeight,
    lineHeight,
    marginLeft,
    marginTop,
    width,
  };

  if (marginY) {
    textStyle.marginTop = marginY;
    textStyle.marginBottom = marginY;
  }
  if (marginX) {
    textStyle.marginLeft = marginX;
    textStyle.marginRight = marginX;
  }

  const textElement = createElement(
    component[variant],
    { className: `typography typography_variant_${variant}`, style: textStyle },
    children,
  );
  return <>{textElement}</>;
}

export default Typography;
