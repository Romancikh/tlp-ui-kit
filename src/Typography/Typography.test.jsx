import { render } from "@testing-library/react";
import Typography from "./Typography";

test("Typography renders correctly", () => {
  const { container } = render(
    <Typography variant="h1" children="h1. Heading" />,
  );
  expect(container).toMatchSnapshot();
});

test("Colored Typography renders correctly", () => {
  const { container } = render(
    <Typography
      color="#1976d2"
      children="body1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam."
    />,
  );
  expect(container).toMatchSnapshot();
});
