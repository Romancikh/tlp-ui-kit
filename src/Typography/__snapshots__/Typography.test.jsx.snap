// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colored Typography renders correctly 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
  >
    body1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
  </p>
</div>
`;

exports[`Typography renders correctly 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
  >
    h1. Heading
  </h1>
</div>
`;
